src/test/java/redis/clients/jedis/tests/commands/AllKindOfValuesCommandsTest.java

```diff
@@ -488,6 +490,23 @@ public void dumpAndRestore() {
     assertTrue(jedis.exists("foo2"));
   }
 
+  @Test(expected = JedisBusyException.class)
+  public void dumpAndRestoreDup() {
+    jedis.set("foo1", "bar1");
+    byte[] sv = jedis.dump("foo1");
+    jedis.restore("foo1", 0, sv);
+    assertTrue(jedis.exists("foo1"));
+  }
+
+  @Test
+  public void dumpAndRestoreReplace() {
+    jedis.set("foo1", "bar1");
+    byte[] sv = jedis.dump("foo1");
+    assertTrue(jedis.exists("foo1"));
+    jedis.restore("foo1", 0, sv, Protocol.DATA_REPLACE);
```

Comment:
I'm not in favor of exposing Protocol to user. It should be a param class which might be similar to SetParam.

https://github.com/xetorthio/jedis/blob/master/src/main/java/redis/clients/jedis/params/set/SetParams.java